apps/addons/models.py

```diff
@@ -1578,7 +1578,9 @@ def version_changed(sender, **kw):
 def update_search_index(sender, instance, **kw):
     from . import tasks
     if not kw.get('raw'):
-        tasks.index_addons.delay([instance.id])
+        if (not instance.disabled_by_user and instance.is_listed and
+                instance.status in amo.VALID_STATUSES):
+            tasks.index_addons.delay([instance.id])
```

Comment:
Nice one ;) It's strange we didn't have this before, does that mean we used to have user disabled or incomplete addons show up in the searches?